miden-crypto/src/merkle/sparse_path.rs

```diff
@@ -0,0 +1,802 @@
+use alloc::vec::Vec;
+use core::{
+    iter::{self, FusedIterator},
+    num::NonZero,
+};
+
+use winter_utils::{Deserializable, DeserializationError, Serializable};
+
+use super::{
+    EmptySubtreeRoots, MerkleError, MerklePath, RpoDigest, SMT_MAX_DEPTH, ValuePath, Word,
+};
+
+/// A different representation of [`MerklePath`] designed for memory efficiency for Merkle paths
+/// with empty nodes.
+///
+/// Empty nodes in the path are stored only as their position, represented with a bitmask. A
+/// maximum of 64 nodes in the path can be empty. The number of empty nodes has no effect on memory
```

Comment:
"The number of empty nodes has no effect on memory" - I think this is wrong? More empty nodes means smaller vector of non-empty nodes?

Reply:
That is a good point! Empty nodes have exactly the opposite effect on memory usage than in a non-sparse path.